db/crud.go

```diff
@@ -1121,15 +1121,38 @@ func (db *Database) resolveDocLocalWins(localDoc *Document, remoteDoc *Document,
 
 	remoteRevID := conflict.RemoteDocument.ExtractRev()
 	remoteGeneration, _ := ParseRevID(remoteRevID)
-	newRevID := CreateRevIDWithBytes(remoteGeneration+1, remoteRevID, docBodyBytes)
+	var newRevID string
 
-	// Set the incoming document's rev and body to the cloned local revision
-	remoteDoc.RevID = newRevID
-	remoteDoc.RemoveBody()
-	remoteDoc._rawBody = docBodyBytes
+	if !localDoc.Deleted {
+		// If the local doc is not a tombstone, we're just rewriting it as a child of the remote
+		newRevID = CreateRevIDWithBytes(remoteGeneration+1, remoteRevID, docBodyBytes)
+	} else {
+		// If the local doc is a tombstone, we're going to end up with both the local and remote branches tombstoned,
+		// and need to ensure the remote branch is the winning branch. To do that, we inject entries into the remote
+		// branch's history until it's generation is longer than the local branch.
+		remoteDoc.Deleted = localDoc.Deleted
+		localGeneration, _ := ParseRevID(localDoc.CurrentRev)
+
+		// TODO: reduce to +1 once CBG-1049 is fixed
+		requiredAdditionalRevs := (localGeneration - remoteGeneration) + 2
```

Comment:
Since we're always adding at least one revision (line 1146), we only need to buffer by an additional 1 revision here (and no buffer required when CBG-1049 is fixed).
```suggestion
		requiredAdditionalRevs := (localGeneration - remoteGeneration) + 1
```

Reply:
Added changes to keep the buffer by an additional 1 revision.